File: common/src/main/java/com.codeabovelab.tpc/text/TextCoordinatesImpl.java
package com.codeabovelab.tpc.text;

import lombok.Value;

/**
 * Default implementation of {@link TextCoordinates }
 */
@Value
public final class TextCoordinatesImpl implements TextCoordinates {
    private final String textualId;
    private final int offset;
    private final int length;
}

File: common/src/main/java/com.codeabovelab.tpc/text/TextCoordinatesImpl.kt
package com.codeabovelab.tpc.text

/**
 * Default implementation of [TextCoordinates]
 */
class TextCoordinatesImpl(
        override val offset: Int,
        override val length: Int)
    : TextCoordinates {
    override fun toString(): String {
        return "TextCoordinatesImpl(offset=$offset, length=$length)"
    }
}
